oeps/oep-0049-django-app-patterns.rst

```diff
@@ -0,0 +1,221 @@
+
+=============================
+OEP-0049: Django App Patterns
+=============================
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-0045 </oeps/oep-0049>`
+   * - Title
+     - Django App Patterns
+   * - Last Modified
+     - 2021-01-29
+   * - Authors
+     - Matt Tuchfarber <mtuchfarber@edx.org>
+   * - Arbiter
+     - Jinder Singh
+   * - Status
+     - Under Review
+   * - Type
+     - Architecture
+   * - Created
+     - 2021-01-29
+   * - Review Period
+     - 2021-02-22 - 2021-03-10
+
+Abstract
+--------
+Proposes a common set of code patterns for Open edX Django apps.
+
+Motivation
+-----------
+As our number of Django apps continue to grow in our many services, we want to coalesce around a couple of standard design patterns to both make switching between codebases easier and to help untangle some of the links between codebases we have today. These decisions should be considered "best practices" or the default patterns, and should only be violated if the code base requires it.
+
+Decision
+--------
+All of our Django apps should have a common structure. This structure consists of a combination of default Django-style code and Open edX-style code. This document will only attempt to detail the common Open edX patterns that we would like to see everywhere, ignoring Django-default items (e.g. ``admin.py``, ``urls.py``, etc) and situation-specific items (e.g. a separate ``constants.py`` file).
```

Comment:
Not a blocking comment but I wasn't able to find a good example of this but it would be nice to link to some docs that explicitly spell out the django side of our layout.  If you're new to django, you won't really know what those are and if you come here for answers, it might be nice to give them a place to go.  The closest I found was this: https://docs.djangoproject.com/en/3.1/intro/tutorial01/#creating-the-polls-app

Reply:
I added a table of the things we won't have opinions on with links to their individual docs. Lemme know if this covers your concern.